src/plonkish/backend/hyperplonk.rs

```diff
@@ -0,0 +1,441 @@
+                        )
+                    })
+                    .collect()
+            })
+            .collect();
+
+        Ok(PlonkishCircuitInfo {
+            k: self.k,
+            num_instances,
+            preprocess_polys,
+            num_witness_polys: self.num_witness_polys.clone(),
+            num_challenges: Default::default(), /* ??? what is challenges used for? This is in
+                                                 * halo2 and the PlonkishCircuitInfo struct but
+                                                 * not in Chiquito */
+            constraints,
+            lookups,
+            permutations: Default::default(), // Chiquito doesn't have permutations
+            max_degree: None,                 // Chiquito doesn't have max degree limit
+        })
+    }
+
+    // preprocessing has three purposes: 1. batch invert fixed assignments, 2. preprocess selectors,
+    // 3. preprocess permutations 1 and 3 don't apply to chiquito; 2 might apply to chiquito but
+    // we choose not to implement it yet therefore, we return the same as
+    // circuit_info_without_preprocess for now
+    fn circuit_info(
+        &self,
+    ) -> Result<plonkish_backend::backend::PlonkishCircuitInfo<F>, plonkish_backend::Error> {
+        let mut circuit_info = self.circuit_info_without_preprocess()?;
+        // make sure all fixed assignments are for fixed column type
+        self.circuit
+            .fixed_assignments
+            .iter()
+            .for_each(|(column, _)| match column.ctype {
+                ColumnType::Fixed => (),
+                _ => panic!("fixed assignments must be for fixed column type"),
+            });
+        // get assignments Vec<F> by looking up from fixed_assignments and reorder assignment
+        // vectors according to self.fixed_uuids finally bind all Vec<F> to a Vec<Vec<F>>
+        // here, filter fixed_assigments: HashMap<Column, Vec<F>> by looking up the Column with uuid
+        let fixed_assignments = self
+            .fixed_uuids
+            .iter()
+            .map(|uuid| {
+                self.circuit
+                    .fixed_assignments
+                    .get(&self.circuit.columns[column_idx(*uuid, &self.all_uuids)])
+                    .unwrap()
+                    .clone()
+            })
+            .collect::<Vec<Vec<F>>>();
+
+        // get selector assignments
+        let selector_assignments_unordered: Assignments<F> = self
+            .assignment_generator
+            .generate_selector_assignments_with_witness(self.trace_witness.clone());
+        let selector_assignments = self
+            .selector_uuids
+            .iter()
+            .map(|uuid| {
+                selector_assignments_unordered
+                    .get(&self.circuit.columns[column_idx(*uuid, &self.all_uuids)])
+                    .unwrap()
+                    .clone()
+            })
+            .collect::<Vec<Vec<F>>>();
```

Comment:
Remove these from the preprocessing phase as they can change between different witnesses.

Reply:
Done.